include/neml2/tensors/BatchTensor.h

```diff
@@ -141,7 +141,7 @@ BatchTensor<N>::BatchTensor(const torch::Tensor & tensor)
 
 template <TorchSize N>
 BatchTensor<N>::BatchTensor(TorchShapeRef batch_size, TorchShapeRef base_size)
-  : torch::Tensor(std::move(torch::zeros(utils::add_shapes(batch_size, base_size), TorchDefaults)))
```

Comment:
When you are reviewing this we should discuss:

I assume you are trying to avoid a copy by using `std::move` here. However, as discussed in this thread https://stackoverflow.com/questions/19267408/why-does-stdmove-prevent-rvo-return-value-optimization, the use of `std::move` in this scenario has quite the opposite effect.

After this change, the regression and verification tests run much faster :)

Reply:
Fun...